### ISSUE TITLE
[DOC] Improve documentation formatting by replacing empty lines with comments

### ISSUE THREAD

**Issue by yarnabrina** [sktime/sktime#5013]
Consider the [documentation](https://www.sktime.net/en/latest/api_reference/auto_generated/sktime.param_est.seasonality.SeasonalityACF.html) of `SeasonalityACF` as an example:

![image](https://github.com/sktime/sktime/assets/39331844/2cccf0c3-27de-4173-8b9e-10caad461c4e)

This is not very readable, and not at all copy-paste friendly.

My guess is this is caused by the new lines in the `Examples` section of the docstring:

https://github.com/sktime/sktime/blob/f5399cd5c6e9839661526d1983f0a223b5da23ec/sktime/param_est/seasonality.py#L58-L85

A way to solve will be to remove the empty line and make the headings into comments.

**Comment by yarnabrina** [comment]
> how we can fix it

I tested locally, and here's what fixes for this example. I don't know how to apply automatically for all examples (or how to find all such cases).

```
     Examples 
     -------- 
     >>> from sktime.datasets import load_airline 
     >>> from sktime.param_est.seasonality import SeasonalityACF 
     >>> 
     >>> X = load_airline().diff()[1:]  # doctest: +SKIP 
     >>> sp_est = SeasonalityACF()  # doctest: +SKIP 
     >>> sp_est.fit(X)  # doctest: +SKIP 
     SeasonalityACF(...) 
     >>> sp_est.get_fitted_params()["sp"]  # doctest: +SKIP 
     12 
     >>> sp_est.get_fitted_params()["sp_significant"]  # doctest: +SKIP 
     array([12, 11]) 
     >>> 
     >>> # Series should be stationary before applying ACF. 
     >>> # To pipeline SeasonalityACF with the Differencer, use the ParamFitterPipeline: 
     >>> from sktime.datasets import load_airline 
     >>> from sktime.param_est.seasonality import SeasonalityACF 
     >>> from sktime.transformations.series.difference import Differencer 
     >>> 
     >>> X = load_airline()  # doctest: +SKIP 
     >>> sp_est = Differencer() * SeasonalityACF()  # doctest: +SKIP 
     >>> sp_est.fit(X)  # doctest: +SKIP 
     ParamFitterPipeline(...) 
     >>> sp_est.get_fitted_params()["sp"]  # doctest: +SKIP 
     12 
     >>> sp_est.get_fitted_params()["sp_significant"]  # doctest: +SKIP 
     array([12, 11]) 
```

Basically, what you can write in python REPL is what seems to be supported by the parser (I hope it's the right term, not sure).

**Comment by fkiraly** [comment]
is there no way to make two consecutive boxes?

**Comment by hazrulakmal** [comment]
based on numpydoc [style guide](https://numpydoc.readthedocs.io/en/stable/format.html#examples) section 15, I quote 

> When multiple examples are provided, they should be separated by blank lines. Comments explaining the examples should have blank lines both above and below them

most of docstrings with multiple examples only have one line above them but not below. A quick example to show that this work is on PR 5065 for `ColumnEnsembleForecaster`

will be great if we know how to automatically detect and edit for all docstring with problematic code example snippets as @yarnabrina suggested. but just incase such automation not possible, I already listed some of them that I stumbled upon in issue 5065 if we wanna do it manually.
